lib/pages/league/widgets/dialogs/rules_dialog.dart

```diff
@@ -0,0 +1,365 @@
+
+class LeagueDialog extends StatefulWidget {
+  const LeagueDialog({super.key});
+
+  @override
+  State<LeagueDialog> createState() => _LeagueDialog();
+}
+
+class _LeagueDialog extends State<LeagueDialog> {
+  final TextEditingController leagueNameController = TextEditingController();
+  final TextEditingController startDateController = TextEditingController();
+  final TextEditingController endDateController = TextEditingController();
+  final TextEditingController teamSizeController = TextEditingController();
+  final TextEditingController participantsController = TextEditingController();
+  final TextEditingController entryFeeController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    leagueNameController.addListener(() {
+      final text = leagueNameController.text.toLowerCase();
+      leagueNameController.value = leagueNameController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+    super.initState();
+    startDateController.addListener(() {
+      final text = startDateController.text.toLowerCase();
+      startDateController.value = startDateController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+    super.initState();
+    endDateController.addListener(() {
+      final text = endDateController.text.toLowerCase();
+      endDateController.value = endDateController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+    super.initState();
+    teamSizeController.addListener(() {
+      final text = teamSizeController.text.toLowerCase();
+      teamSizeController.value = teamSizeController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+    super.initState();
+    participantsController.addListener(() {
+      final text = participantsController.text.toLowerCase();
+      participantsController.value = participantsController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+    super.initState();
+    entryFeeController.addListener(() {
+      final text = entryFeeController.text.toLowerCase();
+      entryFeeController.value = entryFeeController.value.copyWith(
+        text: text,
+        selection:
+            TextSelection(baseOffset: text.length, extentOffset: text.length),
+        composing: TextRange.empty,
+      );
+    });
+  }
```

Comment:
You can remove this. We do not need it.

Also:
In the Flutter widget lifecycle, `initState` should only be called **once**. What you are doing here is calling `initState` several times which is not a good idea.